src/RevProc.cc

```diff
@@ -1680,18 +1708,27 @@ bool RevProc::ClockTick( SST::Cycle_t currentCycle ){
   //
   //
 
-  for( size_t HartID=0; HartID<AssignedThreads.size(); HartID++ ){
-    HART_CTS[HartID] = (AssignedThreads.at(HartID)->GetRegFile()->cost == 0);
+  for( size_t HartID=0; HartID<Harts.size(); HartID++ ){
+    if( HartHasThread(HartID) ){
+      HART_CTS[HartID] = (GetThreadOnHart(HartID)->GetRegFile()->cost == 0);
+    }
+    else {
+      HART_CTS[HartID] = false;
+    }
```

Comment:
I'd just write the whole `if`-`else` statement as:
```c++
 HART_CTS[HartID] = HartHasThread(HartID) && GetThreadOnHart(HartID)->GetRegFile()->cost == 0;
```

Reply:
Well aren't you fancy